src/main.c

```diff
@@ -330,7 +335,12 @@ int main(int argc, char* argv[]) {
         if (further_args_are_strings) {
           ARGS = jv_array_append(ARGS, jv_string(argv[i]));
         } else if (further_args_are_json) {
-          ARGS = jv_array_append(ARGS, jv_parse(argv[i]));
+          jv v =  jv_parse(argv[i]);
+          if (!jv_is_valid(v)) {
+            fprintf(stderr, "%s: invalid JSON text passed to --jsonargs\n", progname);
+            die();
+          }
+          ARGS = jv_array_append(ARGS, v);
```

Comment:
A bit of duplication here. I guess one could extract to some kind of parse_or_die function? could be use for `--argjson` code also. Maybe for later?

Reply:
Nah, it's fine.